### DISCUSSION TITLE
How to recognize a new joined user?

### DISCUSSION THREAD

**Discussion by sohneg** [bwmarrin/discordgo#1186] [Q&A] [answered]
I am new to discordgo and I would like to know, how the bot can recognize a new joined user?

**Comment by JABirchall** [comment, marked as answer]
Hook the GuildMemberAdd event

**Reply by Ekyazed** [reply]
don't work for me :x

**Reply by FedorLap2006** [reply]
To receive this event you must enable `GUILD_MEMBERS` intent before `Open`ing your session:
```go
s.Identify.Intents |= discordgo.IntentGuildMembers
```
However, this intent is priveleged, therefore you also need to enable it on Developer Portal.

**Reply by Ekyazed** [reply]
Oh ! thanks mate

**Comment by databeast** [comment]
If you can't hook the user management Event for some reason, you can still detect joins via the message type when "XYZ just joined the server, hope they bought pizza..." style messages are posted.

```
func ServerJoinMessages(s *discordgo.Session, m *discordgo.MessageCreate) {
	if m.Type == discordgo.MessageTypeGuildMemberJoin {
	     newUserID := m.Author.ID
             ....
	}


```
